alphacsc/tests/test_learn_d_z_multi.py

```diff
@@ -95,74 +101,64 @@ def test_learn_d_z_multi_dicodile(window):
 
 
 @parametrize_solver_and_constraint
-def test_window(rank1, solver_d, uv_constraint):
+@pytest.mark.parametrize('n_trials', [N_TRIALS])
+def test_window(X, rank1, solver_d, uv_constraint):
     # Smoke test that the parameter window does something
-    n_trials, n_channels, n_times = 2, 3, 100
-    n_times_atom, n_atoms = 10, 4
-
-    rng = check_random_state(42)
-    X = rng.randn(n_trials, n_channels, n_times)
-
-    *_, uv_constraint_ = check_solver_and_constraints(
-        rank1, solver_d, uv_constraint
-    )
 
-    D_init = init_dictionary(X, n_atoms, n_times_atom, rank1=rank1,
-                             uv_constraint=uv_constraint_, random_state=0,
+    D_init = init_dictionary(X, N_ATOMS, N_TIMES_ATOM, rank1=rank1,
+                             uv_constraint=uv_constraint, random_state=0,
                              window=False)
 
-    kwargs = dict(X=X, n_atoms=n_atoms, n_times_atom=n_times_atom, verbose=0,
+    kwargs = dict(X=X, n_atoms=N_ATOMS, n_times_atom=N_TIMES_ATOM, verbose=0,
                   uv_constraint=uv_constraint, solver_d=solver_d, rank1=rank1,
                   random_state=0, n_iter=1, solver_z='l-bfgs', D_init=D_init)
     res_False = learn_d_z_multi(window=False, **kwargs)
 
-    D_init = init_dictionary(X, n_atoms, n_times_atom, rank1=rank1,
-                             uv_constraint=uv_constraint_, random_state=0,
+    D_init = init_dictionary(X, N_ATOMS, N_TIMES_ATOM, rank1=rank1,
+                             uv_constraint=uv_constraint, random_state=0,
                              window=True)
 
-    kwargs = dict(X=X, n_atoms=n_atoms, n_times_atom=n_times_atom, verbose=0,
+    kwargs = dict(X=X, n_atoms=N_ATOMS, n_times_atom=N_TIMES_ATOM, verbose=0,
                   uv_constraint=uv_constraint, solver_d=solver_d, rank1=rank1,
                   random_state=0, n_iter=1, solver_z='l-bfgs', D_init=D_init)
+
     res_True = learn_d_z_multi(window=True, **kwargs)
 
     assert not np.allclose(res_False[2], res_True[2])
 
 
-def test_online_learning():
+@pytest.mark.parametrize('n_trials', [N_TRIALS])
```

Comment:
Maybe better to put `N_TRIALS` as a constant by default and only parametrize the test if multiple values are passed?

Reply:
Fixed, see next comment.